homeassistant/components/zha/sensor.py

```diff
@@ -1249,3 +1263,44 @@ class SonoffPresenceSenorIlluminationStatus(Sensor):
     def formatter(self, value: int) -> int | float | None:
         """Numeric pass-through formatter."""
         return SonoffIlluminationStates(value).name
+
+
+@MULTI_MATCH(cluster_handler_names=CLUSTER_HANDLER_THERMOSTAT)
+# pylint: disable-next=hass-invalid-inheritance # needs fixing
+class PiHeatingDemand(Sensor):
+    """Sensor that displays the percentage of heating power used.
+
+    Optional Thermostat attribute
+    """
+
+    _unique_id_suffix = "pi_heating_demand"
+    _attribute_name = "pi_heating_demand"
+    _attr_translation_key: str = "pi_heating_demand"
+    _attr_icon: str = "mdi:radiator"
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _decimals = 0
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+
+
+class SetpointChangeSourceEnum(types.enum8):
+    """The source of the setpoint change."""
+
+    Manual = 0x00
+    Schedule = 0x01
+    External = 0x02
+
+
+@MULTI_MATCH(cluster_handler_names=CLUSTER_HANDLER_THERMOSTAT)
```

Comment:
I think this should rather use `CONFIG_DIAGNOSTIC_MATCH`, as the entity category is set to "diagnostic" below.

Reply:
I looked at other sensor entities in the category diagnostic and those were all matched using MULTI_MATCH. I agree that they probably should be matched with CONFIG_DIAGNOSTIC_MATCH, but maybe that should be done in a separate PR?

Reply:
Uh, weird. That shouldn't be the case I think. Don't update the existing ones, but you can add this at the top:
```python
CONFIG_DIAGNOSTIC_MATCH = functools.partial(
    ZHA_ENTITIES.config_diagnostic_match, Platform.SENSOR
)
```
and only use that for your diagnostic entity.

IIRC there's no real difference between `MULTI_MATCH` and `CONFIG_DIAGNOSTIC_MATCH`, other than that they use a separate registry for the matches and that `MULTI_MATCH` only matches on "unclaimed cluster handlers"(?)

Reply:
Done.

Reply:
> Uh, weird. That shouldn't be the case I think. Don't update the existing ones, but you can add this at the top:
> 
> ```python
> CONFIG_DIAGNOSTIC_MATCH = functools.partial(
>     ZHA_ENTITIES.config_diagnostic_match, Platform.SENSOR
> )
> ```
> 
> and only use that for your diagnostic entity.
> 
> IIRC there's no real difference between `MULTI_MATCH` and `CONFIG_DIAGNOSTIC_MATCH`, other than that they use a separate registry for the matches and that `MULTI_MATCH` only matches on "unclaimed cluster handlers"(?)

Yeah it’s the claiming semantics that are different.